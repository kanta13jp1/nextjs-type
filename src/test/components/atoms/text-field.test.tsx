import TextField from '../../../components/atoms/text-field';
import { TextFieldType } from '../../../data';
import { render } from '../../testUtils';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      push: jest.fn(),
    };
  },
}));

describe('TextField components', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <TextField
        label={'test'}
        type={TextFieldType.Email}
        value={'defalut'}
        classes={['test1', 'test2']}
        error={false}
        onChange={() => console.log('hoge')}
      />,
      {},
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
