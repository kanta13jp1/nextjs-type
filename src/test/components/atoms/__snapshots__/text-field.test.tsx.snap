// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField components matches snapshot 1`] = `
<DocumentFragment>
  <label
    class="block test1 test2"
  >
    <span
      class="text-sm text-gray-700"
    >
      test
    </span>
    <input
      class="mt-1 block w-full rounded-md border-gray-300 focus:border-indigo-600"
      type="email"
      value="defalut"
    />
  </label>
</DocumentFragment>
`;
